storage/bucket.go

```diff
@@ -173,6 +173,9 @@ func (b *BucketHandle) Update(ctx context.Context, uattrs BucketAttrsToUpdate) (
 // [Overview of access control]: https://cloud.google.com/storage/docs/accesscontrol#signed_urls_query_string_authentication
 // [automatic detection of credentials]: https://pkg.go.dev/cloud.google.com/go/storage#hdr-Credential_requirements_for_signing
 func (b *BucketHandle) SignedURL(object string, opts *SignedURLOptions) (string, error) {
+	// Extract the correct endpoint from the readhost set on the client
+	opts.endpoint = b.c.readHost
```

Comment:
I think calling this `endpoint` is a bit misleading. Usually I think of endpoint as being the full path for the JSON API, i.e. `https://storage.googleapis.com/storage/v1`. Just `storage.googleapis.com` would be a hostname.

Sidenote, `readHost` is also a bit misleading because now using XML reads is optional; maybe we should rename to `xmlHost`. (not required for this PR though).

Reply:
You are correct. 

Here is a PR for renaming `readHost`: https://github.com/googleapis/google-cloud-go/pull/8160